web/src/stream_popover.js

```diff
@@ -177,7 +178,7 @@ function build_stream_popover(opts) {
 
             // Choose a different color.
             $popper.on("click", ".choose_stream_color", (e) => {
-                const $popover = $(e.target).closest(".streams_popover");
+                const $popover = $(e.target).closest("#stream-actions-menu-popover");
```

Comment:
Could this be using `instance` to avoid needing to do this kind of lookup?

Reply:
Makes sense to avoid lookups via IDs, when we already have the `instance` object. While redesigning the popovers, I have been using IDs on the popovers since I was working under the assumption that we can have only one instance of a popover at a given time. If we want to keep the flexibility as you have suggested, I think it would make sense for me to use a unique classname instead of an ID on the popover. (Since otherwise, having multiple instances with the same ID would lead to invalid HTML).

Reply:
I have updated the code to now use `$(instance.popper)` instead. Would you want me to put up a small commit which does the above ID -> CLASS conversion?

Reply:
I think it's probably a worthwhile follow-up refactor but very low importance and not worth entangling with this PR, so leave it on your TODO list for potentially a follow-up pass.